Fix tab key handling

SenadAlagic opened the pull request:
I merged the two functions, one for tab indentation and one for autosaving. This seems to work, hopefully this is the implementation you'll be happy with! Cheers!

Fixes #71

SenadAlagic commented:
I see again that there are merge conflicts. This is not the same situation as it was last time? Or do I need to rebase and handle these conflicts myself? Thanks!

waldyrious commented on script.js:
```diff
@@ -9,18 +9,15 @@ calcStats(); // Update counters after loading
-    if (event.keyCode === 9) {
-        event.preventDefault();
+// Auto-save to local storage and calculate stats on every keystroke
+textbox.onkeydown = function (e) {
```
Doesn't `onkeydown` trigger repeatedly if you keep the key pressed down? If so, we'll probably need to use `onkeyup` or `onkeypress` instead to avoid that.

SenadAlagic replied:
For some reason onkeydown is the only one that works. Neither the onkeyup or the onkeypress work, at all. Not sure what's causing it but yes, my original intent was to make it with onkeypress0. The onkeydown does fire repeteadly though. Any ides as to how we can improve it?

waldyrious replied:
I don't really know what might be the issue here, sorry. Perhaps StackOverflow might have some answers?

SenadAlagic replied:
From a quick glance it found this: 

>onkeypress is only triggered for the 'active' field when a key is pressed.
The clue there is the word 'field' -- by default, <div> elements cannot become 'active'; only input fields.

It's late so I will definitely look into this better tomorrow but I have some ideas we could implement. Thanks for the tip!

waldyrious commented on script.js:
```diff
@@ -9,18 +9,15 @@ calcStats(); // Update counters after loading
+// Auto-save to local storage and calculate stats on every keystroke
+textbox.onkeydown = function (e) {
+    if(e.keyCode==9)
+    {
```
Why did you move this bracket to a new line?

SenadAlagic replied:
That's usually how I do it, for my own visibility I guess. Doesn't _really_ matter to me so I'll bring it up.

waldyrious commented on script.js:
```diff
@@ -9,18 +9,15 @@ calcStats(); // Update counters after loading
 function storeLocally() { localStorage.setItem('browserpad', textbox.value); }
 window.beforeunload = storeLocally;
 
-// Allow inputting tabs in the textarea instead of changing focus to the next element
```
This comment should be preserved above the tab-processing code.

SenadAlagic replied:
Will bring it back. Thank you for pointing that out.

SenadAlagic commented:
Hey, just checking up on you. Hopefully you're okay and just busy so you can't respond. I'm ready to continue working whenever you give me the heads up 😄

waldyrious commented:
@SenadAlagic I am so sorry for my lack of response! :broken_heart: I haven't been able to properly maintain my projects and this is totally unfair for contributors. All is good with me, just busier now that I'm a father of two young kids :)

I realize this branch is now conflicted due to my merge of another PR. I'll go ahead and rebase it to fix the conflicts, and hopefully merge it afterwards.

waldyrious committed:
Fix tab key handling

Commit 2e55678411bf changed the tab key handling from onkeydown to onkeypress,
and this broke the functionality. Reverting to onkeydown makes the tab handling work again.

Added an inline comment to explain why this needs to be done with onkeydown.

Co-authored-by: Senad Alagić <senad.alagic@edu.fit.ba>

waldyrious committed:
Improve documentation of onkeyup event handler

waldyrious commented:
@SenadAlagic I actually ended up going with a different approach. I kept the two functions separate, as merging them did not fix this bug, and in fact would have (re)introduced a subtler one that was fixed all the way back in caa8861db0265c8209663afecd826c01c92d6f4c. I added code comments to explain the choices of both `onkeydown` and `onkeyup`.

Thanks again for tackling this and leading the way towards a fix! Merging now.

SenadAlagic commented:
> @SenadAlagic I am so sorry for my lack of response! 💔 I haven't been able to properly maintain my projects and this is totally unfair for contributors. All is good with me, just busier now that I'm a father of two young kids :)
> 
> I realize this branch is now conflicted due to my merge of another PR. I'll go ahead and rebase it to fix the conflicts, and hopefully merge it afterwards.

Hello! Glad to see you're back. Absolutely no need to apologize for spending time with your family. I've been absent from this project for *quite* a while, and I'd like to think I'm more experienced now. If you'd like we can continue working on this, whatever issue or feature you'd like we can surely make it.
